### ISSUE TITLE
Buildbot leaks commits to anonymous users

### ISSUE THREAD

**Issue by Sanqui** [buildbot/buildbot#5212]
Buildbot version: 2.6.0

We are running a Buildbot instance with Github authentication using Authz:

```python
c['www'] = {
    # ...
    'authz': util.Authz(
        allowRules=[
            util.AnyEndpointMatcher(role='role'),
        ],
        roleMatchers=[
            util.RolesFromGroups(),
            util.RolesFromUsername(roles=['role'], usernames=source_access_users),
        ]
    )
}
```

This works perfectly fine for most pages, requiring Github login to see any information.  However the Builds > Last Changes page opens a Websocket which receives information about new commits.  While the page appears empty at first, wait sufficiently long and new commits will crop up.

![image](https://user-images.githubusercontent.com/289437/75465800-71a12800-5989-11ea-94b4-8f37c3b338c6.png)

Assuming this is not a problem with our configuration, I believe this to be a Buildbot issue.

**Comment by tardyp** [comment]
Indeed, the buildbot authorization system is made to protect the rest api, and especially the control rest api.
The websocket api is available with no restriction. you can technically not only watch commit, but also build results and register to any message.

If you want to really restrict read access to the data, you will have to hide your buildbot behind an oauth2_proxy https://github.com/pusher/oauth2_proxy

**Comment by bschatt** [comment]
This is really annoying. Maybe we could add a check in WsResource to ensure the user is logged in before upgrading the connection. This could be an option to avoid breaking compatibility. What do you think @tardyp ?

**Comment by tardyp** [comment]
I think we can add authorization layer in WsResource without breaking compatibility.
You will have to add this to the sse api as well as it is pretty similar.

Obviously any patch allowing to enhance this situation is very welcome.

**Comment by Sanqui** [comment]
The [Buildbot 4.3.0 changelog](https://docs.buildbot.net/latest/relnotes/index.html#buildbot-4-3-0-2025-05-12) features the following bugfix:

>Websocket connections no longer allow unauthenticated users to connect when authorization is enabled

I believe this fixes this issue.  We have since switched to oauth2-proxy as recommended previously though.
